R/data.table.R

```diff
@@ -2532,7 +2531,14 @@ setnames <- function(x,old,new) {
       if (!is.character(old)) stop("'old' is type ",typeof(old)," but should be integer, double or character")
       if (any(duplicated(old))) stop("Some duplicates exist in 'old': ", paste(old[duplicated(old)],collapse=","))
       i = chmatch(old,names(x))
-      if (anyNA(i)) stop("Items of 'old' not found in column names: ",paste(old[is.na(i)],collapse=","))
+      if (anyNA(i)){ if (skip_absent == TRUE){
+        w <- old %chin% names(x)
```

Comment:
what happens when `length(w) == 0`?

Reply:
test 1955.3 now covers that case (all missing)